libs/MeshKernel/src/LandBoundary.cpp

```diff
@@ -12,29 +12,44 @@ void meshkernel::LandBoundary::FindNearestPoint(const Point& samplePoint,
                                                 double& scaledDistanceToStart) const
 {
     nearestPoint = samplePoint;
-    segmentStartIndex = constants::missing::intValue;
+    segmentStartIndex = constants::missing::uintValue;
     scaledDistanceToStart = -1.0;
 
-    // TODO better value for initial value and move to constants namespace
     minimumDistance = 9.0e33;
 
-    for (UInt i = 0; i < m_nodes.size() - 1; ++i)
+    if (samplePoint.IsValid())
```

Comment:
to avoid nesting you could return here?

Reply:
I think here the comparison tool in github has made a mistake

Reply:
If  you are comparing the code 

21 for (Uint i = 0 ...)

to

20 if(samplePoint.IsValid

Reply:
now I see what you mean

Reply:
changed